packages/gamut-labs/src/experimental/AppHeader/__tests__/AppHeader-test.tsx

```diff
@@ -0,0 +1,161 @@
+
+const logoProps: AppHeaderProps = {
+  items: {
+    left: [
+      {
+        id: 'logo-1',
+        href: 'http://codecademy.com',
+        pro: false,
+        trackingTarget: 'tracking-target',
+        type: 'logo',
+      },
+    ],
+    right: [],
+  },
+  onClick: onClick,
+};
+
+const linkProps: AppHeaderProps = {
+  items: {
+    left: [
+      {
+        id: 'link',
+        href: 'http://codecademy.com',
+        trackingTarget: 'tracking-target',
+        text: 'AppHeaderLink',
+        type: 'link',
+      },
+    ],
+    right: [],
+  },
+  onClick: onClick,
+};
+
+const dropdownProps: AppHeaderProps = {
+  items: {
+    left: [
+      {
+        id: 'dropdown',
+        text: 'AppHeaderDropdown',
+        popover: [
+          {
+            id: 'sublink-1',
+            href: 'http://codecademy.com',
+            trackingTarget: 'tracking-target',
+            text: 'AppHeaderLink',
+            type: 'link',
+          },
+          {
+            id: 'sublink-2',
+            href: 'http://codecademy.com',
+            trackingTarget: 'tracking-target',
+            text: 'AppHeaderLink',
+            type: 'link',
+          },
+        ],
+        trackingTarget: 'tracking-target',
+        type: 'dropdown',
+      },
+    ],
+    right: [],
+  },
+  onClick: onClick,
+};
+
+const renderElementProps: AppHeaderProps = {
+  items: {
+    left: [],
+    right: [
+      {
+        id: 'render-element',
+        renderElement: () => <IconButton icon={FaviconIcon} />,
+        type: 'render-element',
+      },
+    ],
+  },
+  onClick: onClick,
+};
+
+const textButtonProps: AppHeaderProps = {
+  items: {
+    left: [],
+    right: [
+      {
+        id: 'text-button',
+        text: 'TextButton',
+        href: 'http://codecademy.com',
+        trackingTarget: 'tracking-target',
+        type: 'text-button',
+      },
+    ],
+  },
+  onClick: onClick,
+};
+
+const fillButtonProps: AppHeaderProps = {
+  items: {
+    left: [],
+    right: [
+      {
+        id: 'fill-button',
+        text: 'FillButton',
+        href: 'http://codecademy.com',
+        trackingTarget: 'tracking-target',
+        type: 'fill-button',
+      },
+    ],
+  },
+  onClick: onClick,
+};
+
+const renderAppHeader = (props: AppHeaderProps) => {
+  return render(
+    <ThemeProvider theme={theme}>
+      <AppHeader {...props} />
+    </ThemeProvider>
+  );
+};
+
+describe('AppHeader', () => {
+  it('renders AppHeaderLogo', () => {
+    renderAppHeader(logoProps);
+    screen.getByTitle('Codecademy Logo');
+  });
+
+  it('renders AppHeaderLink', () => {
+    renderAppHeader(linkProps);
+    screen.getByText('AppHeaderLink');
+  });
+
+  it('renders AppHeaderDropdown', () => {
+    renderAppHeader(dropdownProps);
+    screen.getByText('AppHeaderDropdown');
+  });
+
+  it('renders a custom component', () => {
+    renderAppHeader(renderElementProps);
+    screen.getByTitle('Favicon Icon');
+  });
+
+  it('TextButton calls onClick() when clicked', () => {
```

Comment:
Nit: I think the "it('X when Y')" format is a little easier to skim through for figuring out what tests do. What do you think?

```suggestion
  it('calls onClick when a TextButton is clicked', () => {
```

...and above, for example, `it('renders an AppHeaderLink when the type is link'`)

Reply:
More readable - I like these suggestions!